lib/tic_tac_toe_checker.ex

```diff
@@ -6,8 +6,16 @@ defmodule TicTacToeChecker do
 
   use GenServer
 
+  @type row() :: [non_neg_integer()]
+  @type board() :: [row()]
+  @type cell() :: {non_neg_integer(), non_neg_integer()}
+  @type result() :: [[] | [%{cells: [cell()], player: non_neg_integer()}]]
```

Comment:
Type `[foo()]` includes an empty list. But sometimes it’s even better to specify it explicitly for clarity.

```suggestion
  @type result() :: [[%{cells: [cell()], player: non_neg_integer()}]]
```